xarray/core/common.py

```diff
@@ -155,6 +155,10 @@ def _repr_html_(self):
             return f"<pre>{escape(repr(self))}</pre>"
         return formatting_html.array_repr(self)
 
+    def __format__(self: Any, format_spec: str) -> str:
+        # we use numpy: scalars will print fine and arrays will raise
+        return self.values.__format__(format_spec)
```

Comment:
Should it be `self.data.__format__(format_spec)` instead? What happens with dask arrays for example? Is it ok to trigger a compute for this?

Reply:
Thanks @Illviljan this makes sense. 

> self.data.__format__(format_spec) instead?

All other methods (`__int__`, `__float__`, etc use `.values`, so I assumed the same here.

> What happens with dask arrays for example? 

`.values` triggers a compute, so it works the same. I added a test for this.

> Is it ok to trigger a compute for this?

Good question. I would say yes. `.__format__` really implies that the user wants to convert values to strings, so triggering a compute is ok imo?